samples/scripts/databricks.fsx

```diff
@@ -0,0 +1,29 @@
+#r @"./libs/Newtonsoft.Json.dll"
+#r @"../../src/Farmer/bin/Debug/netstandard2.0/Farmer.dll"
+
+open Farmer
+open Farmer.Builders
+open Farmer.Databricks
+
+let workspace = databricksWorkspace {
+    name "my-databricks-workspace"
+    sku Databricks.Sku.Standard
+    
+    use_public_ip Enabled
+    
+    key_vault "databricks-kv"
```

Comment:
Am I correct that for KeyVault integration to work, you would need to supply all the four KV values (source, name, version and vault)? If so, we should probably feed that through in one of two ways:

1. Force the caller to supply all four values at once (this would also simplify the builder code).
2. At least have some validation in the Builder if they only entered some of the values. (see other builders - they do this in the `Run()` method.

Reply:
I'm not sure what affect using Default or Microsoft.Keyvaults as the KeySource has. The other 3 values do need to be supplied but the version can default to "latest" if its not supplied, otherwise a guid needs to be give. The version should then probably be in the validations.

Reply:
Ok, [this article](https://docs.microsoft.com/en-gb/azure/databricks/security/secrets/secret-scopes) explains that you can either use DataBricks directly for secret management or KeyVault. I've updated the model so that there's no more enum for the secret type. Either you supply the Key Vault configuration or you don't. If you don't, we set the secret type as Default. If you do, we set it as MicrosoftKeyVault. I've also added the "latest" thing you mentioned - where are the docs on that "feature"?

Reply:
Ok I *think* that the encryption is more or less good to go, just need a keyword for databrick encryption. I've rolled back on my thoughts to "implicitly" infer this - there are three states: no encryption at all (is this a good idea?), DataBricks, or KeyVault.

Reply:
I'm not sure that this property refers to the secret scopes. I think it's for encrypting the DBFS root file system, which is a Microsoft managed storage account. It is encrypted by a Microsoft managed key by default but can be encrypted by a customer managed key instead. The details of that can be seen in [this article](https://docs.microsoft.com/en-us/azure/databricks/security/keys/customer-managed-keys-dbfs/)

Reply:
I think I was wrong about using "latest". I must have been thinking of AWS Secrets Manager. It looks like, [from here](https://blog.eldert.net/reference-key-vault-secret-latest-version/), that leaving the version empty gives the latest version instead.

Reply:
I've found some more information on the encryption key properties. It looks like using a secret from a Key Vault requires the "keySource" property to be set to Microsoft.Keyvaults because setting it to Default means that it uses the Microsoft managed key instead. This can be seen [here](https://docs.microsoft.com/en-us/azure/databricks/security/keys/customer-managed-keys-dbfs/cmk-dbfs-azure-cli#disable-customer-managed-keys). With that, there isn't much point allowing any other value than "Microsoft.Keyvaults" for the key source if the key vault, key name or key version are set.

Reply:
1. Ok, so I'll change the `latest` thing to just omit the field.
2. Re: DBFS - ok, that makes sense - I went down the rabbit hole trying to figure that all out. I'll make the naming reflect this.
3. The key source stuff has already been updated to be modelled as you describe - either you will supply the key name for customer-managed (and an optional version) or you'll specify the Databricks (internal managed) option.

Open questions:

* When would you *not* want the `prepareEncryption` field set to true? Or is this that the the case that you don't *have* to encrypt the DBFS stuff?
* Also, how does this interact with the `requireInfrastructureEncryption` in the reference docs?
* There is also a mysterious `storageAccountIdentity` field on the JSON but no real docs as to what it does. Any ideas?

Reply:
(I've just pushed with some renames to keywords and also removed the "latest" in favour of just omitting the field entirely.